cpp/src/lance/io/reader.cc

```diff
@@ -256,7 +270,7 @@ ::arrow::Result<std::shared_ptr<::arrow::RecordBatch>> FileReader::ReadBatch(
 }
 
 int64_t FileReader::GetChunkPosition(int64_t field_id, int64_t chunk_id) const {
-  return lookup_table_->GetOffset(field_id, chunk_id);
+  return lookup_table_->GetOffset(field_id, chunk_id).value_or(-1);
```

Comment:
should we swallow lower level messages here?

Reply:
Yea it is a little bit sloppy to ignore the message here. lemme add some error handling.

Reply:
Fixed.